bundles/org.eclipse.jface.text/projection/org/eclipse/jface/text/source/projection/ProjectionViewer.java

```diff
@@ -518,9 +566,18 @@ public final void disableProjection() {
 	 */
 	public final void enableProjection() {
 		if (!isProjectionMode()) {
+			IRegion visibleRegion= getVisibleRegion();
```

Comment:
this line is not needed here but right before the `if (visibleRegion` block, so please move there.